dask_kubernetes/operator/kubecluster/kubecluster.py

```diff
@@ -60,8 +60,9 @@ class KubeCluster(Cluster):
 
     Parameters
     ----------
-    name: str (required)
-        Name given the Dask cluster.
+    name: str (optional)
```

Comment:
```suggestion
    name: str
```

Setting `name` is required in the vast majority of uses given that `custom_cluster_spec` is a power-user feature. So I'm hesitant to explicitly mark this as `(optional)`. Maybe we just shouldn't specify and leave folks to read the description?